src/cpp/fastdds/publisher/DataWriterImpl.hpp

```diff
@@ -771,6 +773,19 @@ class DataWriterImpl : protected rtps::IReaderDataFilter
 
 private:
 
+    /**
+     * (Re)configures the deadline timer:
+     *  Create once, parked with a huge interval (idle).
+     *  In case of deadline period ∞ cancel it, for 0 warn and notify once; set counts to max and
+     *  for values >0 store period.
+     */
+    void configure_deadline_timer_();
+
+    /**
+     * Notifies listeners that a deadline has been missed without touching the counters.
```

Comment:
I would remove the "without touching the counters" part, if we eventually reset the change counter within the method (still to decide).